Python/pystate.c

```diff
@@ -2197,7 +2197,7 @@ _PyThreadState_PushFrame(PyThreadState *tstate, size_t size)
     assert(size < INT_MAX/sizeof(PyObject *));
     PyObject **base = tstate->datastack_top;
     PyObject **top = base + size;
```

Comment:
Adding size to `NULL` base is undefined behavior. `top` should be calculated after `NULL` check.

Reply:
And similar to the other check, the addition itself might already be undefined behaviour, if it's more than one past the array length.